File: app/src/main/java/alex/com/githubchecker/components/github/repo/list/PRViewHolder.java
package alex.com.githubchecker.components.github.repo.list;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.jakewharton.rxbinding2.view.RxView;

import alex.com.githubchecker.R;
import alex.com.githubchecker.models.api.PullRequest;
import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.subjects.PublishSubject;

/**
 * Created by Alex on 11/13/2017.
 */

class PRViewHolder extends RecyclerView.ViewHolder {

    private View view;

    @BindView(R.id.title) TextView titleTv;
    @BindView(R.id.status) TextView statusTv;

    private Integer _pullRequestNumber;

    PRViewHolder(View itemView, PublishSubject<Integer> selectedPRSubject) {
        super(itemView);
        this.view = itemView;
        ButterKnife.bind(this, view);

        RxView.clicks(view).subscribe(click -> {selectedPRSubject.onNext(_pullRequestNumber);});
    }

    void bind(PullRequest pullRequest) {

        _pullRequestNumber = pullRequest.getNumber();

        String userLogin = pullRequest.getUserLogin();
        Integer number = pullRequest.getNumber();
        String createdAt = pullRequest.getCreatedAt();
        String status = view.getContext().getResources().getString(R.string.pr_content, number, createdAt, userLogin);

        titleTv.setText(pullRequest.getTitle());
        statusTv.setText(status);
    }

}

File: app/src/main/java/alex/com/githubchecker/components/github/repo/list/PRViewHolder.kt
package alex.com.githubchecker.components.github.repo.list

import alex.com.githubchecker.R
import alex.com.githubchecker.models.api.PullRequest
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.jakewharton.rxbinding2.view.RxView
import io.reactivex.subjects.PublishSubject
import kotlinx.android.synthetic.main.item_pr.view.status
import kotlinx.android.synthetic.main.item_pr.view.title

/**
 * Created by Alex on 11/13/2017.
 */

class PRViewHolder(private val containerView: View, selectedPRSubject: PublishSubject<Int>) : RecyclerView.ViewHolder(containerView) {

    private var _pullRequestNumber: Int? = null

    init {
        //@@TODO: dispose
        RxView.clicks(containerView).subscribe { click -> selectedPRSubject.onNext(_pullRequestNumber!!) }
    }

    fun bind(pullRequest: PullRequest) {

        _pullRequestNumber = pullRequest.number

        val userLogin = pullRequest.head?.user?.login
        val number = pullRequest.number
        val createdAt = pullRequest.created_at
        val status = containerView.context.resources.getString(R.string.pr_content, number, createdAt, userLogin)

        containerView.title.text = pullRequest.title
        containerView.status.text = status
    }
}

